reactor-core/src/main/java/reactor/core/publisher/ParallelRunOn.java

```diff
@@ -79,18 +78,32 @@ public void subscribe(CoreSubscriber<? super T>[] subscribers) {
 			Worker w = scheduler.createWorker();
 
 			if (conditional) {
-				parents[i] = new FluxPublishOn.PublishOnConditionalSubscriber<>(
-						(Fuseable.ConditionalSubscriber<T>)subscribers[i],
-						scheduler, w, true,
-						prefetch, prefetch, queueSupplier);
+				parents[i] =
+						new FluxPrefetch.PrefetchSubscriber<>(
```

Comment:
we can keep this, but IIRC the `tryOnNext` path is triggered by sources exclusively, so since `PrefetchSubscriber` isn't conditional it will never be reachable.

Reply:
PublishOn interacts with the downstream by the ConditionalSubscriber contract. Also, PublishOn fuses with the  FluxPrefetch and can simply poll values from its queue.

Reply:
Also, PublishOn doesn't implement the ConditionalSubscriber interface and I can't replace PrefetchSubscriber with PrefetchConditionalSubscriber.

Reply:
ok got it. `PublishOnConditionalSubscriber` naming is a bit confusing, but I don't really have an idea for a better name to convey that it _uses_ the downstream as a `ConditionalSubscriber` 🤔